src/sql/SqlService.ts

```
@@ -0,0 +1,290 @@
+
+        try {
+            const serializedParams = [];
+            if (Array.isArray(sqlStatement.params)) { // params can be undefined
+                for (let param of sqlStatement.params) {
+                    param = this.convertToStringIfDatetimeValue(param);
+                    serializedParams.push(this.serializationService.toData(param));
+                }
+            }
+            const cursorBufferSize = sqlStatement.options?.cursorBufferSize ?
+                sqlStatement.options.cursorBufferSize : SqlServiceImpl.DEFAULT_CURSOR_BUFFER_SIZE;
+            const requestMessage = SqlExecuteCodec.encodeRequest(
+                sqlStatement.sql,
+                serializedParams,
+                sqlStatement.options?.timeoutMillis ? sqlStatement.options.timeoutMillis : SqlServiceImpl.DEFAULT_TIMEOUT,
+                cursorBufferSize,
+                sqlStatement.options?.schema ? sqlStatement.options.schema : SqlServiceImpl.DEFAULT_SCHEMA,
+                expectedResultType,
+                queryId
+            );
+
+            const res = SqlResultImpl.newResult(
+                this,
+                this.serializationService,
+                connection,
+                queryId,
+                cursorBufferSize,
+                sqlStatement.options?.hasOwnProperty('returnRawResult') ?
+                    sqlStatement.options.returnRawResult : SqlServiceImpl.DEFAULT_FOR_RETURN_RAW_RESULT,
+                this.connectionManager.getClientUuid()
+            );
+
+            this.invocationService.invokeOnConnection(connection, requestMessage).then(clientMessage => {
+                this.handleExecuteResponse(clientMessage, res);
+            }).catch(err => {
+                console.log(err);
+                res.onExecuteError(
+                    new HazelcastSqlException(
+                        connection.getRemoteUuid(), SqlErrorCode.CONNECTION_PROBLEM, err.message, err
```

Would it be to check whether or not the connection is alive here? If it is not alive, we can say that cluster topology is changed (like the Java side), and if it is, we can throw a generic error